src/sonic-config-engine/portconfig.py

```diff
@@ -51,4 +70,98 @@ def parse_port_config_file(port_config_file):
             port_alias_map[data['alias']] = name
     return (ports, port_alias_map)
 
+def parse_platform_json_file(port_config_file):
+    ports = {}
+    port_alias_map = {}
+
+    # Read 'platform.json' file
+    try:
+        with open(port_config_file) as fp:
+            try:
+                data = json.load(fp)
+            except json.JSONDecodeError:
+                print("Json file does not exist")
+        port_dict = ast.literal_eval(json.dumps(data))
+    except:
+        print("error occurred while parsing json:", sys.exc_info()[1])
 
+    """ Regex for  SYMMETRIC breakout_modes
+            Syntax  : number x speed1 [speed2, speed3]
+                    : means the parent port lanes are split equally to number of ports denoted by the number. And the speed for each breakout port is default to speed1, but is changeable to speed2, speed3 etc.
+    """
+    SYMMETRIC_SPLIT_BREAKOUT_PATTERN = "(^\d{1})x(\d{2,3}G)(\[\d{2,3}G\])?"
+
+    """ Regex for ASYMMETRIC breakout_modes
+            Syntax  : number1 x speed1[speed2,..](numberA) + number2 x speedX[speedY,..](numberB) + ...
```

Comment:
Can we change the Syntax in comments to the below?

Syntax: N1XS1[S2, S3, .. Sn]
N1 --> Number of child ports created, lanes are split equally among child ports
S1 --> Default speed of child ports.
Si --> Supported speed for child ports, where i = 2..n

Reply:
I have kept the syntax as per the HLD document. But, yes, we can modify it as you suggested. Let's discuss this.